**Documentation**:

*   we have a total of **962** sartup
*   **columns** are: name, url , class, status , description
*   we have 153 startups without url ==> 15.9%
*   Dropped url column because it's redundant
*   None of our class column is missing values
*   For all of our 962 startups we have only 24 class
*   77.75% of our data is missing status
*   we have 2 status for startups : exited/ dead/nan
*   we replaced nan values with -1 for the startup status column and exited with 1 and dead with 0
*   we have 9 missing startups description; we're manually filling them
*   we have 5 startups names that are being duplicated but they appear to not be the same check below for more details 
* **TEXT CLEANING**:
*   For description column: Removed punctuations+stop words+lowercase+removed diacritics
+ Profile Reporter





In [64]:
import pandas as pd
import numpy as np
import texthero as hero
from pandas_profiling import ProfileReport

### We need to open the csv with an encoding of ***latin1*** else we'll face an error.

In [65]:
df=pd.read_csv("/home/seneca/Documents/Omdena/pennyslvenia_startupsucces/omdena-pennsylvania-startup/src/tasks/task-1-data-collection/YCombinatorCompanies.csv", encoding="latin1")
df.head()


Name                          URL Class Status  \
0       Awesound         https://awesound.com    F2    NaN   
1   BillionToOne  http://www.billiontoone.com    F2    NaN   
2       Bulletin     http://shop-bulletin.com    F2    NaN   
3        Cerplus    http://www.getcerplus.com    F2    NaN   
4  ConnectHealth  http://www.connecthealth.io    F2    NaN   

                                        Description  
0                              AdWords for podcasts  
1                    Fetal genetic testing in India  
2                                 Airbnb for retail  
3               B2B marketplace for surplus produce  
4  Telemedicine and prescription delivery in Africa

# Column 1: Name cleaning

In [66]:
unique_name=len(df.Name.unique())
all_names= len(df.Name)
print(all_names) , print(unique_name)

962
957


(None, None)

### we have 5 names that are being repeated, we need to check them
1. See if they're duplicted
2. If they're 2 startups with the same name
3. If they updated one of their status

In [67]:
df[df.duplicated(subset=['Name']) == True ]

Name                          URL Class Status  \
124     Restocks          https://restocks.io   W16    NaN   
132     Shypmate      http://www.shypmate.com   W16    NaN   
138  StrongIntro                          NaN   W16   Dead   
338       Supply  http://www.madebysupply.com   W15    NaN   
461        Weave      http://www.getweave.com   W14    NaN   

                                           Description  
124  Push notifications for superconsumers. We make...  
132  Shypmate enables a fast, reliable and cost eff...  
138  StrongIntro helps recruiters turn their entire...  
338                    Wireless power, through the air  
461  Better phone service for your dental practice ...



  ## Restocks

1.  Class difference :  F1 vs W16
2.   Description 




In [68]:
df[df["Name"]== "Restocks"]

Name                  URL Class Status  \
40   Restocks  https://restocks.io    F1    NaN   
124  Restocks  https://restocks.io   W16    NaN   

                                           Description  
40   Push notifications when sold out products restock  
124  Push notifications for superconsumers. We make...



##  Shypmate


1.  Class difference :  F1 vs W16
2.  Description 





In [69]:
df[df["Name"]== "Shypmate"]

Name                      URL Class Status  \
41   Shypmate  http://www.shypmate.com    F1    NaN   
132  Shypmate  http://www.shypmate.com   W16    NaN   

                                           Description  
41   Affordable and fast international peer to peer...  
132  Shypmate enables a fast, reliable and cost eff...



##   StrongIntro
1.  Class difference :  F1 vs W16
2.   Description 
3. Status : Nan vs Dead


In [70]:
df[df["Name"]== "StrongIntro"]

Name                      URL Class Status  \
43   StrongIntro  https://strongintro.com    F1    NaN   
138  StrongIntro                      NaN   W16   Dead   

                                           Description  
43   Recruiting tool and service that 10xÕs your em...  
138  StrongIntro helps recruiters turn their entire...



##   Supply
1.  Class difference :  S15 vs W15



In [71]:
df[df["Name"]== "Supply"]

Name                          URL Class Status  \
236  Supply  http://www.madebysupply.com   S15    NaN   
338  Supply  http://www.madebysupply.com   W15    NaN   

                         Description  
236  Wireless power, through the air  
338  Wireless power, through the air



##   Weave
1.  Two different startups : not the same description


In [72]:
df[df["Name"]== "Weave"]

Name                      URL Class Status  \
406  Weave                      NaN   S14   Dead   
461  Weave  http://www.getweave.com   W14    NaN   

                                           Description  
406  Professional networking to arrange meetings an...  
461  Better phone service for your dental practice ...

### Didn't drop any of them because we can get insights from the change of Class status

# Column 2: URLS

In [73]:
print(df["URL"].isna().sum())
df[df["URL"].isna() == True]

153


Name  URL Class  Status  \
29           Convrge  NaN    F1    Dead   
33        FlightMate  NaN    F1    Dead   
35   Jetty Financial  NaN    F1    Dead   
123        Redspread  NaN   W16    Dead   
138      StrongIntro  NaN   W16    Dead   
..               ...  ...   ...     ...   
953            YouOS  NaN   W06    Dead   
954       ClickFacts  NaN   S05     NaN   
955         Infogami  NaN   S05  Exited   
958           Memamp  NaN   S05    Dead   
960          Simmery  NaN   S05    Dead   

                                           Description  
29            Hang out with friends in virtual reality  
33     Flexible flight querying with automated booking  
35                Virtual assistant to manage finances  
123          Version control for software environments  
138  StrongIntro helps recruiters turn their entire...  
..                                                 ...  
953  YouOS provides a platform for developers to cr...  
954  ClickFacts is a Malware Software Solutions com...  
955  Merged with Reddit.ÊInfogami is a wiki applica...  
958                   A better desktop search solution  
960                                                NaN  

[153 rows x 5 columns]

## we have 153 startups without Url ==> 15.9%
Website is a redundant feature and will not give us any insight, we better drop the column

In [74]:
df.drop(columns=["URL"], axis=1, inplace=True)
df.head()

Name Class Status  \
0       Awesound    F2    NaN   
1   BillionToOne    F2    NaN   
2       Bulletin    F2    NaN   
3        Cerplus    F2    NaN   
4  ConnectHealth    F2    NaN   

                                        Description  
0                              AdWords for podcasts  
1                    Fetal genetic testing in India  
2                                 Airbnb for retail  
3               B2B marketplace for surplus produce  
4  Telemedicine and prescription delivery in Africa

# Column 3: Class

In [75]:
print(df["Class"].isna().sum())

0


### Good thing, ,none of our startups has a missing status value

In [76]:
class_list=df["Class"].unique()
class_list.sort()
print(len(class_list) ,class_list )

24 ['F1' 'F2' 'S05' 'S06' 'S07' 'S08' 'S09' 'S10' 'S11' 'S12' 'S13' 'S14'
 'S15' 'W06' 'W07' 'W08' 'W09' 'W10' 'W11' 'W12' 'W13' 'W14' 'W15' 'W16']


### For all of our 962 startups we have only 24 class


# Column 4 : Status

In [77]:
df["Status"].isna().sum()

748

## 77.75% of our data is missing status

In [78]:
df["Status"].unique()

array([nan, 'Dead', 'Exited'], dtype=object)

In [79]:
dead_status= df[df["Status"]== 'Exited']
len(dead_status)

76

In [80]:
exited_status= df[df["Status"]== 'Exited']
len(exited_status)

76

From the remaining data : 

1.   76 starups have the exited status
2.   138 startups have the dead status



# Column 5 : Description

In [81]:
df.head()

Name Class Status  \
0       Awesound    F2    NaN   
1   BillionToOne    F2    NaN   
2       Bulletin    F2    NaN   
3        Cerplus    F2    NaN   
4  ConnectHealth    F2    NaN   

                                        Description  
0                              AdWords for podcasts  
1                    Fetal genetic testing in India  
2                                 Airbnb for retail  
3               B2B marketplace for surplus produce  
4  Telemedicine and prescription delivery in Africa

In [82]:
df["Description"].isna().sum()

9

we only have 9 startups without description

In [83]:
startup_missing_descri= df[df["Description"].isna() == True]
startup_missing_descri

Name Class Status Description
778     I Move You   S10   Dead         NaN
788      Simperium   S10    NaN         NaN
819      Tweetflow   W10   Dead         NaN
832         Jobpic   S09   Dead         NaN
875     Precognate   S08   Dead         NaN
882  UrbanTakeover   S08   Dead         NaN
931          View3   W07   Dead         NaN
947      AudioBeta   W06    NaN         NaN
960        Simmery   S05   Dead         NaN

## Replacing nan values with -1 + dead with 0+ exited with 1

In [84]:
df.fillna(-1, inplace=True)
df["Status"].replace({"Exited": "1", "Dead": "0"}, inplace=True)

In [85]:
df

Name Class Status  \
0         Awesound    F2     -1   
1     BillionToOne    F2     -1   
2         Bulletin    F2     -1   
3          Cerplus    F2     -1   
4    ConnectHealth    F2     -1   
..             ...   ...    ...   
957          Loopt   S05      1   
958         Memamp   S05      0   
959         Reddit   S05      1   
960        Simmery   S05      0   
961      TextPayMe   S05      1   

                                           Description  
0                                 AdWords for podcasts  
1                       Fetal genetic testing in India  
2                                    Airbnb for retail  
3                  B2B marketplace for surplus produce  
4     Telemedicine and prescription delivery in Africa  
..                                                 ...  
957  Acquired by Green Dot for $43.4MM.ÊLoopt is a ...  
958                   A better desktop search solution  
959  Acquired by Conde Nast in 2006.ÊA social news ...  
960                                                 -1  
961  Acquired by Amazon for ~$3MM.ÊAn SMS payment s...  

[962 rows x 4 columns]

In [86]:
##TEXT CLEANING
# let's do text preprocessing
from texthero import preprocessing

# creating a custom pipeline to preprocess the raw text we have
custom_pipeline = [preprocessing.fillna
                   , preprocessing.lowercase
                   , preprocessing.remove_punctuation
                   , preprocessing.remove_diacritics
                   , preprocessing.remove_stopwords
                   , preprocessing.remove_whitespace
                   ]
# simply call clean() method to clean the raw text in 'tweet' col and pass the custom_pipeline to pipeline argument
df['clean_Description'] = hero.clean(df['Description'], pipeline = custom_pipeline)

df.tail()

Name Class Status  \
957      Loopt   S05      1   
958     Memamp   S05      0   
959     Reddit   S05      1   
960    Simmery   S05      0   
961  TextPayMe   S05      1   

                                           Description  \
957  Acquired by Green Dot for $43.4MM.ÊLoopt is a ...   
958                   A better desktop search solution   
959  Acquired by Conde Nast in 2006.ÊA social news ...   
960                                                 -1   
961  Acquired by Amazon for ~$3MM.ÊAn SMS payment s...   

                                     clean_Description  
957  acquired green dot 43 4mm eloopt social servic...  
958                     better desktop search solution  
959  acquired conde nast 2006 ea social news websit...  
960                                                  1  
961  acquired amazon 3mm ean sms payment service le...

In [87]:
df.to_csv("cleanedY_combinator_data22.csv")

In [ ]:
profile= ProfileReport(df, title="Startup Dataset", html={'style':{'full_width':True}})
profile.to_notebook_iframe()